govc/library/export.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vapi/library"
+	"github.com/vmware/govmomi/vapi/library/finder"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vim25/soap"
+)
+
+type export struct {
+	*flags.ClientFlag
+	*flags.OutputFlag
+	library.Item
+}
+
+func init() {
+	cli.Register("library.export", &export{})
+}
+
+func (cmd *export) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClientFlag, ctx = flags.NewClientFlag(ctx)
+	cmd.ClientFlag.Register(ctx, f)
+
+	cmd.OutputFlag, ctx = flags.NewOutputFlag(ctx)
+	cmd.OutputFlag.Register(ctx, f)
+}
+
+func (cmd *export) Usage() string {
+	return "ITEM"
+}
+
+func (cmd *export) Description() string {
+	return `Export library items.
+
+Examples:
+  govc library.export library_name/item_name
+  govc library.export library_name/item_name/file_name
+  govc library.export library_name/item_name/*.ovf -`
+}
+
+func (cmd *export) Process(ctx context.Context) error {
+	if err := cmd.ClientFlag.Process(ctx); err != nil {
+		return err
+	}
+	return cmd.OutputFlag.Process(ctx)
+}
+
+func (cmd *export) Run(ctx context.Context, f *flag.FlagSet) error {
+	return cmd.WithRestClient(ctx, func(c *rest.Client) error {
```

Comment:
Possible improvement would be to add a comment about the various modes and calling conventions. Looks like there can be a number of names specified, items/files, and along with a "-" for stdout.

Reply:
Good call @markpeek , I've updated the docs.